script.js

```diff
@@ -1,4 +1,160 @@
+    return `LEADERBOARD <br><br> Player 1 Score: ${player1Score} <br><br> Player 2 Score: ${player2Score}<br><br> Press Enter to start another round.`;
+  } else {
+    return `LEADERBOARD <br><br> Player 2 Score: ${player2Score} <br><br> Player 1 Score: ${player1Score}<br><br> Press Enter to start another round.`;
+  }
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  if (currentMode == "start") {
+    myOutputValue = `Choose normal or reversed mode to begin 🎲`;
+  }
+  if (currentMode == "start" && input == "normal") {
+    myOutputValue = `You have chosen NORMAL mode. <br><br> Press enter to start rolling the dice 🎲`;
+    currentMode = "normal mode";
+  } else if (currentMode == "start" && input == "reversed") {
+    myOutputValue = `You have chosen REVERSED mode. <br><br> Press enter to start rolling the dice 🎲`;
+    currentMode = "normal mode";
+    reversedMode = true;
+    // normal mode will roll the dice
+  } else if (currentMode == "normal mode") {
+    diceRoll1Player1 = diceRollGenerator();
+    diceRoll2Player1 = diceRollGenerator();
+    myOutputValue = player1RollDice();
+    currentMode = "player 1 rolled";
+    // input the order
+  } else if (currentMode == "player 1 rolled" && input == 1) {
+    var player1Order = diceOrder1Player1();
+    currentMode = "player 2 start";
+    return player1Order;
+  } else if (currentMode == "player 1 rolled" && input == 2) {
+    var player1Order = diceOrder2Player1();
+    currentMode = "player 2 start";
+    return player1Order;
+
+    // player 2 roll the dice
+  } else if (currentMode == "player 2 start") {
+    diceRoll1Player2 = diceRollGenerator();
+    diceRoll2Player2 = diceRollGenerator();
+    myOutputValue = player2RollDice();
+    currentMode = "player 2 rolled";
+  } else if (currentMode == "player 2 rolled" && input == 1) {
+    var player2Order = diceOrder1Player2();
+    currentMode = "results";
+    return player2Order;
+  } else if (currentMode == "player 2 rolled" && input == 2) {
+    var player2Order = diceOrder2Player2();
+    currentMode = "results";
+    return player2Order;
+
+    // shows result
+  } else if (currentMode == "results" && player1FinalNumb == player2FinalNumb) {
+    myOutputValue = `Its a draw. <br><br> Player 1: ${player1FinalNumb} <br><br> Player 2: ${player2FinalNumb}`;
+    currentMode = "leaderboard";
+  } else if (
+    currentMode == "results" &&
+    player1FinalNumb > player2FinalNumb &&
+    reversedMode == false
+  ) {
+    player1Score++;
+    currentMode = "leaderboard";
+    myOutputValue = `Player 1 wins!<br><br> Player 1: ${player1FinalNumb} <br><br> Player 2: ${player2FinalNumb}`;
+  } else if (
+    currentMode == "results" &&
+    player1FinalNumb < player2FinalNumb &&
+    reversedMode == false
+  ) {
+    player2Score++;
+    myOutputValue = `Player 2 wins!<br><br> Player 1: ${player1FinalNumb} <br><br> Player 2: ${player2FinalNumb}`;
```

Comment:
Ideally when your code is in the same game mode, it means that the code can potentially fit into one game mode.
The above code would be better if you placed them in a helper function like determineWinner(), which then checks for the various win conditions.

This is similar to the previous 2 comments where when it was in "player 1 rolled" / "player 2 rolled" mode, and I suggested that they could be combined into one function